assets/src/stories-editor/blocks/amp-story-page-attachment/attachment-content.js

```diff
@@ -61,45 +62,45 @@ const AttachmentContent = ( props ) => {
 		const { getSettings } = select( 'amp/story' );
 		const { allowedPageAttachmentPostTypes } = getSettings();
 		return allowedPageAttachmentPostTypes;
-	} );
+	}, [] );
 
-	const fetchSelectedPost = () => {
+	const fetchSelectedPost = useCallback( () => {
 		const restBase = allowedPostTypes[ postType ] || `${ postType }s`;
 
-		isStillMounted = true;
+		isStillMounted.current = true;
 		if ( postId ) {
 			setIsFetching( true );
-			const currentFetchRequest = fetchRequest = apiFetch( {
+			const currentFetchRequest = fetchRequest.current = apiFetch( {
 				path: `/wp/v2/${ restBase }/${ postId }`,
 			} ).then(
 				( post ) => {
-					if ( isStillMounted && fetchRequest === currentFetchRequest ) {
+					if ( isStillMounted.current && fetchRequest.current === currentFetchRequest ) {
 						setSelectedPost( post );
 						setFailedToFetch( false );
 						setIsFetching( false );
 					}
 				}
 			).catch(
 				() => {
-					if ( isStillMounted && fetchRequest === currentFetchRequest ) {
+					if ( isStillMounted.current && fetchRequest.current === currentFetchRequest ) {
 						setSelectedPost( null );
 						setFailedToFetch( true );
 						setIsFetching( false );
 					}
 				}
 			);
 		}
-	};
+	}, [ postId, postType, allowedPostTypes ] );
```

Comment:
Since `allowedPostTypes` is not something that the user can actually change, is it necessary to track the change here?

Reply:
There's no benefit in omitting it, plus this helps to prevent issues in case this should ever change in the future.